src/platform/packages/shared/kbn-server-route-repository/src/test_types.ts

```diff
@@ -372,6 +464,27 @@ client
     }>(res);
   });
 
+client
+  .fetch('GET /internal/endpoint_with_response_validation_zod', {
+    params: {
+      query: {
+        start: '',
+      },
+    },
+    timeout: 1,
+  })
+  .then((res) => {
+    assertType<{
+      success: true;
+      data: Array<{ id: number }>;
+    }>(res);
+
+    assertType<{
+      success: false;
+      error: string;
+    }>(res);
```

Comment:
This likely passes because indeed the type can be inferred to this but I don't believe the response validation for the 400 is running and since we're not actually returning a 400, this ends up in `then` while a real 400 should end up in a rejection.

Reply:
Yes, it shouldn't be 400. I've updated the test to use 2xx as currently error responses aren't unwrapped as with `.ok` response.